drivers/flash/soc_flash_numaker.c

```diff
@@ -0,0 +1,281 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2023 Nuvoton Technology Corporation.
+ */
+
+#define DT_DRV_COMPAT nuvoton_numaker_fmc
+
+#include <string.h>
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+#include "flash_priv.h"
+#include <NuMicro.h>
+
+LOG_MODULE_REGISTER(flash_numaker, CONFIG_FLASH_LOG_LEVEL);
+
+#define SOC_NV_FLASH_NODE DT_INST(0, soc_nv_flash)
+#define SOC_NV_FLASH_WRITE_BLOCK_SIZE DT_PROP_OR(SOC_NV_FLASH_NODE, write_block_size, 0x04)
+
+struct flash_numaker_data {
+	FMC_T *fmc;
+	struct k_sem write_lock;
+	uint32_t flash_block_base;
+};
+
+static const struct flash_parameters flash_numaker_parameters = {
+	.write_block_size = SOC_NV_FLASH_WRITE_BLOCK_SIZE,
+	.erase_value = 0xff,
+};
+
+/* Validate offset and length */
+static bool flash_numaker_is_range_valid(off_t offset, size_t len)
+{
+	/* check for min value */
+	if ((offset < 0) || (len < 1)) {
```

Comment:
`len < 1` is redundant check as `size_t` is unsigned and the only possible value to trigger this would be `len == 0`, something that callers of the function check before calling this funciton anywa.

Reply:
Thanks and resolved